bin/unwrap_snaphu.py

```diff
@@ -0,0 +1,666 @@
+      Path to scratch directory
+    power: isce3.io.gdal.Raster (or None)
+      Slc power raster in the reference SLC geometry
+    mask: isce3.io.gdal.Raster (or None)
+      Binary mask to apply during unwrapping (gdal.GDT_Byte)
+    unw_est: isce3.io.gdal.Raster (or None)
+      Coarse unwrapped phase estimate (gdal.GDT_Float32)
+    """
+    # Get cost mode and corresponding cost parameter object
+    # assign deformation if not found
+    snaphu_cfg = clean_none_cfg_keys(snaphu_cfg)
+    cost_mode = snaphu_cfg.get('cost_mode', 'defo')
+    cost_params = select_cost_options(snaphu_cfg, cost_mode)
+
+    # Get tiling parameters
+    tile_params = set_tile_params(snaphu_cfg)
+    # Get solver parameters
+    solver_params = set_solver_params(snaphu_cfg)
+    # Get connected components parameters
+    conn_comp_params = set_connected_components_params(snaphu_cfg)
+    # Get correlation bias model parameters
+    corr_model_params = set_corr_bias_params(snaphu_cfg)
+    # Get phase standard deviation model parameters
+    phase_std_params = set_phase_std_params(snaphu_cfg)
+
+    verbose = snaphu_cfg['verbose'] if snaphu_cfg.get(
+        'verbose') is not None else False
+    debug = snaphu_cfg['debug'] if snaphu_cfg.get(
+        'debug') is not None else False
+
+    # All parameters are set. Ready to unwrap
+    snaphu.unwrap(ugram, conn_comp, igram, corr,
+                  nlooks, cost_mode, cost_params, power,
+                  mask, unw_est, tile_params, solver_params,
+                  conn_comp_params, corr_model_params, phase_std_params,
+                  scratchdir, verbose, debug)
+
+
+def main(cfg):
+    """
+    Main function. cfg: snaphu dictionary
+    """
+
+    # Check that input files are valid
+    check_input_files(cfg)
+
+    # Check output directory
+    if (cfg.get('scratchdir') is not None) and (
+            not os.path.exists(cfg.get('scratchdir'))):
+        os.mkdir(cfg['scratchdir'], exist_ok=True)
+
+    # Loop over interferograms to unwrap
+    for k in range(len(cfg['igram_raster'])):
+        # Put interferogram and correlation in a ISCE3 raster
+        igram_raster = isce3.io.gdal.Raster(cfg['igram_raster'][k])
+        corr_raster = isce3.io.gdal.Raster(cfg['correlation_raster'][k])
+
+        # If ugram_raster and connected components are None use igram basename
+        if cfg['unwrapped_raster'] is None:
+            basename = os.path.basename(cfg['igram_raster'][k])
+            ugram_name = f'{basename}.ugram'
+            conn_comp_name = ugram_name.replace('ugram', 'conn_comp')
+        else:
+            ugram_name = cfg['unwrapped_raster'][k]
+            conn_comp_name = cfg['connected_components_raster'][k]
+
+        # Create unwrapped and connected components rasters.
+        ugram_raster = isce3.io.gdal.Raster(ugram_name, igram_raster.width,
+                                            igram_raster.length, np.float32,
+                                            "GTiff")
```

Comment:
Either make the default ENVI so we can see the results with MDX or make the format configurable with default to ENVI.